### PR TITLE
chore(BATS): Public file tests

### PR DESCRIPTION
**Pull request by agsuy**
Some tests that assert existence of files.

### CODE REVIEW

File: bats_test/downloads/file-downloads/publlic_files.bats [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@test 'Verifies "Coffee_chart.jpg" file exists' {
+    assert_exist "/home/node/ardrive-cli/bats_test/downloads/file-downloads/Coffee_chart.jpg"
+}
+@test 'Verifies "samplee.mkv" file exists' {
+    assert_exist "/home/node/ardrive-cli/bats_test/downloads/file-downloads/samplee.mkv"
+}
+@test 'Verifies "test!_·$%d\%&·_.txt" file exists' {
+    #BATS parsing issue with that filename, always fails
+    skip
+    assert_exist "/home/node/ardrive-cli/bats_test/downloads/file-downloads/test!_·\$%d\%&·_.txt"
```
Comment (matibat) [inline bats_test/downloads/file-downloads/publlic_files.bats:38]: If the issue here is regarding the special characters of the file name, then using single quotes would do the trick:
```suggestion
    assert_exist '/home/node/ardrive-cli/bats_test/downloads/file-downloads/test!_·\$%d\%&·_.txt'
```

File: bats_test/downloads/file-downloads/publlic_files.bats [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# File methods. DEPENDS on SUPPORT lib.
+load '/home/node/packages/node_modules/bats-file/load.bash'
+# Assertions.  DEPENDS on SUPPORT lib.
+load '/home/node/packages/node_modules/bats-assert/load.bash'
+
+DIR="$(cd "$(dirname "$BATS_TEST_FILENAME")" >/dev/null 2>&1 && pwd)"
+
+setup_file() {
+    run $DIR/pre_download.sh
+    echo "##### Downloaded 7 files" >>/home/node/bats.log
```
Comment (matibat) [inline bats_test/downloads/file-downloads/publlic_files.bats:14]: What log should we dump here? Custom log messages?

File: bats_test/downloads/file-downloads/pre_download.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    "d3399761-ece8-4404-b560-6822472c9f62"
+    "366b951f-c3cd-4136-b086-faecd7872394"
+    "1571ccbd-7be8-4040-98df-0e36f8ffacc9"
+    "438f4c26-0e0d-42d1-b821-11109170a794"
+)
+
+for publicFile in ${myFiles[@]}; do
+    yarn ardrive download-file -f $publicFile
+done
+exit 1
```
Comment (matibat) [inline bats_test/downloads/file-downloads/pre_download.sh:16]: Why'd to exit with `1` here? Shouldn't it return `0`?

File: bats_test/downloads/file-downloads/publlic_files.bats [hunk 4]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bats
+
+# Support lib
+load '/home/node/packages/node_modules/bats-support/load.bash'
+# File methods. DEPENDS on SUPPORT lib.
+load '/home/node/packages/node_modules/bats-file/load.bash'
+# Assertions.  DEPENDS on SUPPORT lib.
+load '/home/node/packages/node_modules/bats-assert/load.bash'
+
+DIR="$(cd "$(dirname "$BATS_TEST_FILENAME")" >/dev/null 2>&1 && pwd)"
```
Comment (matibat) [inline bats_test/downloads/file-downloads/publlic_files.bats:10]: This line is:

 - `cd` into the parent of `${BATS_TEST_FILENAME}`
   - the `stdout` of `cd` is redirected to `/dev/null`
   - the `stderr` is redirected to `stdout`
     - if there was an error doing `cd`, then returns `pwd` to put in `DIR`
     - if `cd` had success, then that's the value to put in `DIR`
-------------------
So, is there a case where `${BATS_TEST_FILENAME}` has a wrong value?

Comment (matibat) [reply bats_test/downloads/file-downloads/publlic_files.bats:10]: Can't we have a hardcoded `${BATS_TEST_BASENAME}`?

File: bats_test/downloads/file-downloads/publlic_files.bats [hunk 5]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bats
```
Comment (matibat) [inline bats_test/downloads/file-downloads/publlic_files.bats:1]: **TODO** rename this file to `public_files.bats`

**Review by matibat — COMMENTED** [review]
It's such a good example!

Left a couple of comments

File: bats_test/downloads/file-downloads/publlic_files.bats (hunk 4 above)
Comment (agsuy) [reply bats_test/downloads/file-downloads/publlic_files.bats:10]: This way the test ll always work as it fetches the absolute path dynamically

File: bats_test/downloads/file-downloads/publlic_files.bats (hunk 2 above)
Comment (agsuy) [reply bats_test/downloads/file-downloads/publlic_files.bats:14]: This is how you create log files according to BATS docs, yes

File: bats_test/downloads/file-downloads/publlic_files.bats (hunk 5 above)
Comment (agsuy) [reply bats_test/downloads/file-downloads/publlic_files.bats:1]: Fixed

File: bats_test/downloads/file-downloads/pre_download.sh (hunk 3 above)
Comment (agsuy) [reply bats_test/downloads/file-downloads/pre_download.sh:16]: True. Fixed.

File: bats_test/downloads/file-downloads/public_files.bats [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+DIR="$(cd "$(dirname "$BATS_TEST_FILENAME")" >/dev/null 2>&1 && pwd)"
+
+setup_file() {
+    run $DIR/pre_download.sh
+    echo "##### Downloaded 7 files" >>/home/node/bats.log
+}
+
+@test 'Verifies "1 space test.zip" file exists' {
+    assert_exist "$(pwd)/1 space test.zip"
```
Comment (matibat) [inline bats_test/downloads/file-downloads/public_files.bats:18]: Shouldn't we use the `DIR` variable here?
```suggestion
    assert_exist "${DIR}/1 space test.zip"
```

Comment (agsuy) [reply bats_test/downloads/file-downloads/public_files.bats:18]: Noup. Stuff is downloaded in an external script whereas test asserts relative to where it is located.

**Comment by matibat** [comment]
Approved. My only concern is we have a dynamic path for the files being downloaded. I'd like to see them in the same folder always, but tet's discuss that behavior later on

**Pull request merged**